src/entities/swap/swaps/v3/index.ts

```diff
@@ -120,17 +123,31 @@ export class SwapV3 implements SwapBase {
             client,
         });
         if ('exactAmountIn' in this.swaps) {
-            const { result } =
-                await routerContract.simulate.querySwapSingleTokenExactIn(
-                    [
-                        this.swaps.pool,
-                        this.swaps.tokenIn,
-                        this.swaps.tokenOut,
-                        this.swaps.exactAmountIn,
-                        DEFAULT_USERDATA,
-                    ],
-                    { blockNumber: block },
-                );
+            const { result } = account
+                ? // pseudo query fails if account has insufficient allowance or balance
```

Comment:
Don't have a big preference on ternary vs another method but did find it pretty tricky to notice the `?` with the comment inline like this.